panic when running `kubectl diff` on malformed spec

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

If the matter is security related, please disclose it privately via https://kubernetes.io/security/
-->


**What happened**:
I tried to `kubectl diff` with a Deployment spec that was roughly shaped like this (note that the `imagePullSecrets` is invalid, since it should be a list of `LocalObjectReference` instead of `string`). This caused kubectl to fail to create a three-way patch and panic.
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: foo
  namespace: foo
spec:
  template:
    spec:
      imagePullSecrets:
        - foo-auth
```

<details>
<summary>Click here for the stacktrace</summary>

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x18 pc=0x4e4dbe]

goroutine 1 [running]:
fmt.Fprintf(0x0, 0x0, 0x1b6e8a8, 0x37, 0xc0000bcca0, 0x1, 0x1, 0xdf1, 0xdf1, 0x1e0df20)
        /usr/local/go/src/fmt/print.go:200 +0x7e
k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/apply.(*Patcher).patchSimple(0xc0008bab00, 0x1de9fa0, 0xc0008c0000, 0xc002392180, 0xba, 0xba, 0xc00055d800, 0x9, 0xc000e1a13a, 0x6, ...)
        /workspace/anago-v1.16.0-rc.2.1+2bd9643cee5b3b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/apply/apply.go:915 +0xe7f
k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/apply.(*Patcher).Patch(0xc0008bab00, 0x1de9fa0, 0xc0008c0000, 0xc002392180, 0xba, 0xba, 0xc00055d800, 0x9, 0xc000e1a13a, 0x6, ...)
        /workspace/anago-v1.16.0-rc.2.1+2bd9643cee5b3b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/apply/apply.go:959 +0xf3
k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/diff.InfoObject.Merged(0x1de9fa0, 0xc0007b12e0, 0xc000e18540, 0x1ddf320, 0xc0008ae5d0, 0x1dddd40, 0xc000dc96a0, 0x0, 0xc0025b4230, 0x19350a0, ...)
        /workspace/anago-v1.16.0-rc.2.1+2bd9643cee5b3b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/diff/diff.go:328 +0x3c4
k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/diff.(*DiffVersion).getObject(0xc000e11d00, 0x1e0e160, 0xc000e3a0c0, 0x1dde480, 0xc0008c0010, 0x0, 0x0)
        /workspace/anago-v1.16.0-rc.2.1+2bd9643cee5b3b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/diff/diff.go:194 +0x153
k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/diff.(*DiffVersion).Print(0xc000e11d00, 0x1e0e160, 0xc000e3a0c0, 0x0, 0x0)
        /workspace/anago-v1.16.0-rc.2.1+2bd9643cee5b3b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/diff/diff.go:201 +0x52
k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/diff.(*Differ).Diff(0xc000dd4a60, 0x1e0e160, 0xc000e3a0c0, 0xc000e3a0c0, 0x2)
        /workspace/anago-v1.16.0-rc.2.1+2bd9643cee5b3b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/diff/diff.go:373 +0x82
k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/diff.(*DiffOptions).Run.func1(0xc000e18540, 0x0, 0x0, 0x0, 0x0)
        /workspace/anago-v1.16.0-rc.2.1+2bd9643cee5b3b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/diff/diff.go:497 +0x2ed
k8s.io/kubernetes/vendor/k8s.io/cli-runtime/pkg/resource.DecoratedVisitor.Visit.func1(0xc000e18540, 0x0, 0x0, 0xc0024e3b60, 0xc002666600)
        /workspace/anago-v1.16.0-rc.2.1+2bd9643cee5b3b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/cli-runtime/pkg/resource/visitor.go:335 +0xe0
k8s.io/kubernetes/vendor/k8s.io/cli-runtime/pkg/resource.FlattenListVisitor.Visit.func1(0xc000e18540, 0x0, 0x0, 0x0, 0x0)
        /workspace/anago-v1.16.0-rc.2.1+2bd9643cee5b3b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/cli-runtime/pkg/resource/visitor.go:399 +0x9b2
k8s.io/kubernetes/vendor/k8s.io/cli-runtime/pkg/resource.FlattenListVisitor.Visit.func1(0xc000e18540, 0x0, 0x0, 0xb0, 0xc00055d800)
        /workspace/anago-v1.16.0-rc.2.1+2bd9643cee5b3b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/cli-runtime/pkg/resource/visitor.go:399 +0x9b2
k8s.io/kubernetes/vendor/k8s.io/cli-runtime/pkg/resource.(*StreamVisitor).Visit(0xc000db39c0, 0xc000db3a80, 0x1000, 0xc000280000)
        /workspace/anago-v1.16.0-rc.2.1+2bd9643cee5b3b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/cli-runtime/pkg/resource/visitor.go:597 +0x281
k8s.io/kubernetes/vendor/k8s.io/cli-runtime/pkg/resource.(*FileVisitor).Visit(0xc000e11d40, 0xc000db3a80, 0x0, 0x0)
        /workspace/anago-v1.16.0-rc.2.1+2bd9643cee5b3b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/cli-runtime/pkg/resource/visitor.go:528 +0x20b
k8s.io/kubernetes/vendor/k8s.io/cli-runtime/pkg/resource.VisitorList.Visit(0xc000dd4ad0, 0x1, 0x1, 0xc000db3a80, 0xc000db3a01, 0xc000db3a80)
        /workspace/anago-v1.16.0-rc.2.1+2bd9643cee5b3b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/cli-runtime/pkg/resource/visitor.go:197 +0x6a
k8s.io/kubernetes/vendor/k8s.io/cli-runtime/pkg/resource.FlattenListVisitor.Visit(0x1ddf700, 0xc000e11d60, 0x1decb60, 0x312c2a0, 0xc000df8fc0, 0xc000db3a40, 0xc000693b38, 0xc000693ad0)
        /workspace/anago-v1.16.0-rc.2.1+2bd9643cee5b3b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/cli-runtime/pkg/resource/visitor.go:391 +0x8d
k8s.io/kubernetes/vendor/k8s.io/cli-runtime/pkg/resource.FlattenListVisitor.Visit(0x1ddf6c0, 0xc000df8ff0, 0x1decb60, 0x312c2a0, 0xc000df8fc0, 0xc000db3a00, 0x19b8800, 0xc000693b80)
        /workspace/anago-v1.16.0-rc.2.1+2bd9643cee5b3b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/cli-runtime/pkg/resource/visitor.go:391 +0x8d
k8s.io/kubernetes/vendor/k8s.io/cli-runtime/pkg/resource.DecoratedVisitor.Visit(0x1ddf6c0, 0xc000df9050, 0xc000e11da0, 0x3, 0x4, 0xc000e11dc0, 0xc000e11dc0, 0xc000e11d80)
        /workspace/anago-v1.16.0-rc.2.1+2bd9643cee5b3b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/cli-runtime/pkg/resource/visitor.go:326 +0x8d
k8s.io/kubernetes/vendor/k8s.io/cli-runtime/pkg/resource.(*Result).Visit(0xc000d73700, 0xc000e11dc0, 0xc0006e7680, 0xc000095a40)
        /workspace/anago-v1.16.0-rc.2.1+2bd9643cee5b3b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/cli-runtime/pkg/resource/result.go:99 +0x69
k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/diff.(*DiffOptions).Run(0xc0006e7680, 0x0, 0x0)
        /workspace/anago-v1.16.0-rc.2.1+2bd9643cee5b3b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/diff/diff.go:461 +0x186
k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/diff.NewCmdDiff.func1(0xc0006e8f00, 0xc0001d9ec0, 0x0, 0x3)
        /workspace/anago-v1.16.0-rc.2.1+2bd9643cee5b3b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/diff/diff.go:114 +0xb3
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).execute(0xc0006e8f00, 0xc0001d9e90, 0x3, 0x3, 0xc0006e8f00, 0xc0001d9e90)
        /workspace/anago-v1.16.0-rc.2.1+2bd9643cee5b3b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:830 +0x2b5
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc000552500, 0x1bf4588, 0xc0000c4000, 0x5)
        /workspace/anago-v1.16.0-rc.2.1+2bd9643cee5b3b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:914 +0x303
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).Execute(...)
        /workspace/anago-v1.16.0-rc.2.1+2bd9643cee5b3b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:864
main.main()
        _output/dockerized/go/src/k8s.io/kubernetes/cmd/kubectl/kubectl.go:49 +0x1ee
```
</details>

**What you expected to happen**:
An error is returned

**How to reproduce it (as minimally and precisely as possible)**:
Create a deployment, then try to diff using the YAML provided above (make sure to replace the name and namespace)

**Anything else we need to know?**:
The panic occurs on this line because `errOut` is nil:
https://github.com/kubernetes/kubernetes/blob/2bd9643cee5b3b3a5ecbd3af49d09018f0773c77/staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go#L915

It's set to nil here:
https://github.com/kubernetes/kubernetes/blob/2bd9643cee5b3b3a5ecbd3af49d09018f0773c77/staging/src/k8s.io/kubectl/pkg/cmd/diff/diff.go#L328

**Environment**:
- Kubernetes version (use `kubectl version`):  
```
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.0", GitCommit:"2bd9643cee5b3b3a5ecbd3af49d09018f0773c77", GitTreeState:"clean", BuildDate:"2019-09-18T14:36:53Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"windows/amd64"}
```
- Cloud provider or hardware configuration: N/A
- OS (e.g: `cat /etc/os-release`): N/A
- Kernel (e.g. `uname -a`): N/A
- Install tools: N/A
- Network plugin and version (if this is a network-related bug): N/A
- Others: N/A

--- a/staging/src/k8s.io/kubectl/pkg/cmd/diff/diff.go
+++ b/staging/src/k8s.io/kubectl/pkg/cmd/diff/diff.go
@@ -257,6 +257,7 @@ type InfoObject struct {
 	Force           bool
 	ServerSideApply bool
 	ForceConflicts  bool
+	genericclioptions.IOStreams
 }
 
 var _ Object = &InfoObject{}
@@ -325,7 +326,7 @@ func (obj InfoObject) Merged() (runtime.Object, error) {
 		ResourceVersion: resourceVersion,
 	}
 
-	_, result, err := patcher.Patch(obj.Info.Object, modified, obj.Info.Source, obj.Info.Namespace, obj.Info.Name, nil)
+	_, result, err := patcher.Patch(obj.Info.Object, modified, obj.Info.Source, obj.Info.Namespace, obj.Info.Name, obj.ErrOut)
 	return result, err
 }
 
@@ -492,6 +493,7 @@ func (o *DiffOptions) Run() error {
 				Force:           force,
 				ServerSideApply: o.ServerSideApply,
 				ForceConflicts:  o.ForceConflicts,
+				IOStreams:       o.Diff.IOStreams,
 			}
 
 			err = differ.Diff(obj, printer)

